modules/test-utils/src/snapshot-test-runner.ts

```diff
@@ -140,8 +141,20 @@ export class SnapshotTestRunner extends TestRunner<
     };
     // Take screenshot and compare
     const result = await window.browserTestDriver_captureAndDiffScreen(diffOptions);
+
+    // If failed, try if we have a platform specific golden image
+    let resultOverride;
+    const platform = this.testOptions.imageDiffOptions?.platform;
+    if (!result.success) {
+      diffOptions.goldenImage = diffOptions.goldenImage.replace(
+        'golden-images/',
+        `golden-images/platform-overrides/${platform}/`
+      );
+      resultOverride = await window.browserTestDriver_captureAndDiffScreen(diffOptions);
+    }
```

Comment:
Minor - Would we prefer to have it compare to the platform-specific golden image on the first try? If something changes and both platforms render the same image, we otherwise won't know anything has changed, and the platform-specific golden image would be left behind unused in the folder.

Reply:
Annoyingly I couldn't figure out a way to detect if a platform specific image exists, otherwise I would do this. Given that we expect the platform overrides to only be a few images (hopefully!) I don't want to slow down all the tests by needlessly running them for non-existent images